### PR TITLE
Modified project to work with doctest instead of catch

### PR DESCRIPTION
**Pull request by carlosvin**
I have chosen doctest to be in master because it is supposed to be faster and lighter: 

- https://github.com/onqtam/doctest/blob/master/doc/markdown/benchmarks.md#cost-of-including-the-header
- https://github.com/onqtam/doctest/blob/master/doc/markdown/faq.md#how-is-doctest-different-from-catch

### CODE REVIEW

File: test/uuid_test.cpp [hunk 1]
```diff
@@ -1,42 +1,38 @@
...
   for (int i=0; i<MAX_ITERS; i++) {
     ids::Uuid uuid;
     std::string uuid_str {uuid.to_str()};
 
-    INFO(uuid_str);
-    REQUIRE(uuid.most > 0);
-    REQUIRE(uuid.least > 0);
-    REQUIRE(uuid_str.size() == 36);
+    MESSAGE(uuid_str);
+    CHECK(uuid.most > 0);
```
Comment (carlosvin) [inline test/uuid_test.cpp:15]: CHECK in Doctest is equivalent to REQUIRE in Catch

File: test/uuid_test.cpp [hunk 2]
```diff
@@ -1,42 +1,38 @@
...
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
 
+#include "doctest.h"
 #include "Uuid.h"
-#include "catch.hpp"
 #include <string>
 
 constexpr int MAX_ITERS = 100;
 
-TEST_CASE( "Uuid", "[uuid]" ) {
```
Comment (carlosvin) [inline test/uuid_test.cpp:14]: It seems in Doctest is not allowed to add labels to test case

File: test/uuid_test.cpp [hunk 3]
```diff
@@ -1,42 +1,38 @@
...
-    REQUIRE(uuid_str.size() == 36);
+    MESSAGE(uuid_str);
+    CHECK(uuid.most > 0);
+    CHECK(uuid.least > 0);
+    CHECK(uuid_str.size() == 36);
   }
 }
+// SECTION in catch = SUBCASE in doctest
 
 // BDD style
```
Comment (carlosvin) [inline test/uuid_test.cpp:22]: BDD style use same template names in both frameworks: GIVEN, WHEN, THEN

File: test/uuid_test.cpp (hunk 1 above)
Comment (onqtam) [reply test/uuid_test.cpp:15]: Hmmm... CHECK should have the same semantics in both frameworks - and REQUIRE too (REQUIRE terminates the test case with a throw statement when it fails)

Comment (carlosvin) [reply test/uuid_test.cpp:15]: Understood, I will modify examples using either CHECK or REQUIRE in both (Catch or doctest). Thanks!

**Pull request merged**